<documentation title="Cast Structure Spacing">
    <standard>
    <![CDATA[
    Surround type casts with spaces.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: space after typecast.">
        <![CDATA[
$a = (int) '420';
        ]]>
        </code>
        <code title="Invalid: space after typecast.">
        <![CDATA[
$a = (int)'420';
        ]]>
        </code>
    </code_comparison>
</documentation>
