<documentation title="Disallow Inline Tabs">
    <standard>
    <![CDATA[
    Use spaces for inline alignments.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: spaces used for inline alignment.">
        <![CDATA[
$a = array(
  'lorem'[space]=> 'ipsum',
  'dol'[space][space][space]=> 'or',
);
        ]]>
        </code>
        <code title="Invalid: tabs used for inline alignment.">
        <![CDATA[
$a = array(
  'lorem'[tab]=> 'ipsum',
  'dol'[tab]=> 'or',
);
        ]]>
        </code>
    </code_comparison>
</documentation>
