<documentation title="Operator Spacing">
    <standard>
    <![CDATA[
    Always put a space after commas, and on both sides of logical, comparison, string and assignment operators.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: one space before and after the operator.">
        <![CDATA[
if ( $a<em> === </em>$b<em> && </em>$b<em> === </em>$c ) {}
if (<em> ! </em>$var ) {}
        ]]>
        </code>
        <code title="Invalid: too much/little space.">
        <![CDATA[
// Too much space.
if ( $a === $b<em>   &&   </em>$b<em> ===      </em>$c ) {}
if (<em>  ! </em>$var ) {}

// Too little space.
if ( $a<em>===</em>$b<em> &&</em>$b<em> ===</em>$c ) {}
if (<em> !</em>$var ) {}
        ]]>
        </code>
    </code_comparison>
    <code_comparison>
        <code title="Valid: multiline is okay, too.">
        <![CDATA[
if ( $a === $b<em>
    && </em>$b === $c
) {}
        ]]>
        </code>
		<code title="Invalid: too much space after operator on new line.">
        <![CDATA[
if ( $a === $b<em>
    &&     </em>$b === $c
) {}
        ]]>
        </code>
    </code_comparison>
</documentation>
