<documentation title="Precision Alignment">
    <standard>
    <![CDATA[
    Line indentation should only use tabs. Precision alignment with spaces after the tabs is not strongly discouraged.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: only tabstops used for indentation.">
        <![CDATA[
[tab]$var = true;
        ]]>
        </code>
        <code title="Invalid: precision alignment of 2 spaces.">
        <![CDATA[
[space][space]$var = true;
        ]]>
        </code>
    </code_comparison>
    <code_comparison>
        <code title="Valid: four spaces counts as a tab, the replacement of these will be handled by another sniff.">
        <![CDATA[
[tab][space][space][space][space]$var = true;
        ]]>
        </code>
        <code title="Invalid: precision alignment of 3 spaces.">
        <![CDATA[
[tab][space][space][space]$var = true;
        ]]>
        </code>
    </code_comparison>
</documentation>
