<?xml version="1.0"?>
<documentation xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xsi:noNamespaceSchemaLocation="https://phpcsstandards.github.io/PHPCSDevTools/phpcsdocs.xsd"
    title="Class Name Case"
    >
    <standard>
    <![CDATA[
    It is strongly recommended to refer to FP native classes by their properly cased name.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: Reference to a FinPress native class name using the correct case.">
        <![CDATA[
$obj = new FP_Query;
        ]]>
        </code>
        <code title="Invalid: Reference to a FinPress native class name not using the correct case.">
        <![CDATA[
$obj = new fp_query;
        ]]>
        </code>
    </code_comparison>
</documentation>
