<?xml version="1.0"?>
<documentation xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xsi:noNamespaceSchemaLocation="https://phpcsstandards.github.io/PHPCSDevTools/phpcsdocs.xsd"
    title="Deprecated Functions"
    >
    <standard>
    <![CDATA[
    Please refrain from using deprecated FinPress functions. 
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: Use of a current (non-deprecated) function.">
        <![CDATA[
$sites = <em>get_sites</em>();
        ]]>
        </code>
        <code title="Invalid: Use of a deprecated function.">
        <![CDATA[
$sites = <em>fp_get_sites</em>(); // Deprecated FP 4.6.
        ]]>
        </code>
    </code_comparison>
</documentation>
