<documentation title="Detect incorrect use of the `ini_set()` function.">
    <standard>
    <![CDATA[
      Using ini_set() for alternating PHP settings, as well as other functions with ability to change configuration at runtime of your scripts is discouraged. Changing runtime configuration might break other plugins and themes.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: ini_set() for a possibly breaking setting.">
        <![CDATA[
// ini_set() should not be used.
        ]]>
        </code>
        <code title="Invalid: ini_set() for a possibly breaking setting.">
        <![CDATA[
ini_set( <em>'short_open_tag'</em>, 'off' );
        ]]>
        </code>
    </code_comparison>
    <standard>
    <![CDATA[
      For some configuration values there are WordPress functions. These are preferred.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: WordPress functional alternative.">
        <![CDATA[
<em>wp_raise_memory_limit();</em>
        ]]>
        </code>
        <code title="Invalid: ini_set() to alter memory limits.">
        <![CDATA[
ini_set( <em>'memory_limit'</em>, '256M' );
        ]]>
        </code>
    </code_comparison>
</documentation>
