<?xml version="1.0"?>
<documentation title="Type Casts">
    <standard>
    <![CDATA[
      This sniff normalizes float type cast keywords.

      Use (float) instead of legacy synonyms:
      - Disallowed: (double), (real)
      - Allowed:    (float)

      Note: Other type-cast rules in the WordPress standard
      are enforced by different sniffs:
      - Short forms like (int), (bool), (string):
        PSR12.Keywords.ShortFormTypeKeywords
      - No spaces inside parentheses:
        Squiz.WhiteSpace.CastSpacing
      - Exactly one space after the cast:
        Generic.Formatting.SpaceAfterCast
    ]]>
    </standard>

    <code_comparison>
        <code title="Invalid: legacy float casts">
        <![CDATA[
<?php
$price = <em>(double)</em> $value;
$size  = <em>(real)</em>   $value;
        ]]>
        </code>
        <code title="Valid: normalized float cast">
        <![CDATA[
<?php
$price = <em>(float)</em> $value;
$size  = <em>(float)</em> $value;
        ]]>
        </code>
    </code_comparison>
</documentation>
