<documentation title="Array Key Spacing Restrictions">
    <standard>
    <![CDATA[
      When referring to array items, only include a space around the index if it is a variable.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: Correct whitespace for index keys">
        <![CDATA[
$post       = $posts[ $post_id ];
$post       = $posts[123];
$post_title = $post['post_title'];
        ]]>
        </code>
        <code title="Invalid: Incorrect spacing">
        <![CDATA[
$post       = $posts[$post_id];
$post       = $posts[ 123 ];
$post_title = $post[ 'post_title' ];
        ]]>
        </code>
    </code_comparison>
</documentation>
