<?xml version="1.0"?>
<documentation xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xsi:noNamespaceSchemaLocation="https://phpcsstandards.github.io/PHPCSDevTools/phpcsdocs.xsd"
    title="Valid Taxonomy Slug"
    >
    <standard>
    <![CDATA[
    The taxonomy slug used in register_taxonomy() must be between 1 and 32 characters.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: Short taxonomy slug.">
        <![CDATA[
register_taxonomy(
    <em>'my_short_slug'</em>,
    array()
);
        ]]>
        </code>
        <code title="Invalid: Too long taxonomy slug.">
        <![CDATA[
register_taxonomy(
    <em>'my_very_own_taxonomy_is_much_too_long'</em>,
    array()
);
        ]]>
        </code>
    </code_comparison>
    <standard>
    <![CDATA[
    The taxonomy slug used in register_taxonomy() can only contain lowercase alphanumeric characters, dashes and underscores.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: No special characters in taxonomy slug.">
        <![CDATA[
register_taxonomy(
    <em>'my_taxonomy_slug'</em>,
    array()
);
        ]]>
        </code>
        <code title="Invalid: Invalid characters in taxonomy slug.">
        <![CDATA[
register_taxonomy(
    <em>'my/taxonomy/slug'</em>,
    array()
);
        ]]>
        </code>
    </code_comparison>
    <standard>
    <![CDATA[
    One should be careful with passing dynamic slug names to "register_taxonomy()", as the slug may become too long and could contain invalid characters.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: Static taxonomy slug.">
        <![CDATA[
register_taxonomy(
    <em>'my_taxonomy_active'</em>,
    array()
);
        ]]>
        </code>
        <code title="Invalid: Dynamic taxonomy slug.">
        <![CDATA[
register_taxonomy(
    <em>"my_taxonomy_{$status}"</em>,
    array()
);
        ]]>
        </code>
    </code_comparison>
    <standard>
    <![CDATA[
    The taxonomy slug used in register_taxonomy() can not use reserved names, such as the ones used by WordPress itself.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: Prefixed taxonomy slug.">
        <![CDATA[
register_taxonomy(
    <em>'prefixed_post_tag'</em>,
    array()
);
        ]]>
        </code>
        <code title="Invalid: Using a reserved keyword as slug.">
        <![CDATA[
register_taxonomy(
    <em>'post_tag'</em>,
    array()
);
        ]]>
        </code>
    </code_comparison>
    <standard>
    <![CDATA[
    The taxonomy slug used in register_taxonomy() can not use reserved prefixes, such as 'wp_', which is used by WordPress itself.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: Custom prefix taxonomy slug.">
        <![CDATA[
register_taxonomy(
    <em>'prefixed_post_tag'</em>,
    array()
);
        ]]>
        </code>
        <code title="Invalid: Using a reserved prefix.">
        <![CDATA[
register_taxonomy(
    <em>'wp_post_tag'</em>,
    array()
);
        ]]>
        </code>
    </code_comparison>
</documentation>
